core/src/main/java/org/codice/imaging/nitf/core/common/NitfDateTime.java

```diff
@@ -80,10 +82,13 @@ public final String getSourceString() {
         This is a best-effort, based on the information available, which can be incomplete. If
         no other information is available, a default Date will be returned.
 
-        @return the value of this object as a Date.
-    */
+        @return the value of this object as a Date, or null if the date is not valid.
+     */
```

Comment:
The source string is important for round-tripping files. Its a little cost for significantly improved testability, so would like to keep it. 

I can live without Date, but kept it around for interfacing with older APIs / joda-time.